File: frontend/src/app/model/currency.ts
export class Currency {
  id: number;
  shortName: string;
  fullName: string;
  country: string;

  constructor(shortName: string, fullName: string, country: string) {
    this.shortName = shortName;
    this.fullName = fullName;
    this.country = country;
  }
}

File: src/main/java/com/project/app/model/Currency.java
package com.project.app.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "currencies")
public class Currency {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Size(min = 3, max = 3)
    private String shortName;

    @NotNull
    @Size(max = 30)
    private String fullName;

    @NotNull
    @Size(max = 30)
    private String country;
}
